cmd/zdb/zdb.c

```diff
@@ -7049,7 +7087,23 @@ main(int argc, char **argv)
 			return (error);
 		} else {
 			zdb_set_skip_mmp(target);
-			error = open_objset(target, FTAG, &os);
+			if (dataset_lookup == B_TRUE) {
+				/*
+				 * Use the supplied id to get the name
+				 * for open_objset.
+				 */
+				error = spa_open(target, &spa, FTAG);
+				if (error == 0) {
+					error = name_from_objset_id(spa,
+					    objset_str, dsname);
+					spa_close(spa, FTAG);
+					if (error == 0)
```

Comment:
What do you think about removing this block and falling though to the existing `open_objset` block below with the now known dataset name.